src/web/web-service/wpan_service.cpp

```diff
@@ -74,13 +74,16 @@ std::string WpanService::HandleJoinNetworkRequest(const std::string &aJoinReques
     }
 
     VerifyOrExit(client.FactoryReset(), ret = kWpanStatus_LeaveFailed);
-    VerifyOrExit(client.Execute("masterkey %s", masterKey.c_str()) != nullptr, ret = kWpanStatus_SetFailed);
-    VerifyOrExit(client.Execute("networkname %s", mNetworks[index].mNetworkName) != nullptr,
+    VerifyOrExit(client.Execute("dataset init new") != nullptr, ret = kWpanStatus_SetFailed);
+    VerifyOrExit(client.Execute("dataset masterkey %s", masterKey.c_str()) != nullptr, ret = kWpanStatus_SetFailed);
+    VerifyOrExit(client.Execute("dataset networkname %s", mNetworks[index].mNetworkName) != nullptr,
```

Comment:
Not caused by this PR, but here we didn't escape whitespaces in the network name.

Reply:
fine to address in a separate PR.

Reply:
It may not have too much to do with `ot-ctl`. 
For example, `ot-ctl` can sets the network name to "hello world" by 
```bash
$ ot-ctl "networkname hello\ world"
```

Reply:
Created #578 to track this.